controllers/release/release_adapter.go

```diff
@@ -38,22 +39,24 @@ import (
 
 // Adapter holds the objects needed to reconcile a Release.
 type Adapter struct {
-	release *v1alpha1.Release
-	logger  logr.Logger
-	client  client.Client
-	context context.Context
+	release       *v1alpha1.Release
+	logger        logr.Logger
+	client        client.Client
+	context       context.Context
+	targetContext context.Context
 }
 
 // finalizerName is the finalizer name to be added to the Releases
 const finalizerName string = "appstudio.redhat.com/release-finalizer"
 
 // NewAdapter creates and returns an Adapter instance.
-func NewAdapter(release *v1alpha1.Release, logger logr.Logger, client client.Client, context context.Context) *Adapter {
+func NewAdapter(release *v1alpha1.Release, logger logr.Logger, client client.Client, context, targetContext context.Context) *Adapter {
```

Comment:
Don't allow to pass a `targetContext` if this is something we will never use (and we won't as we have an operation to take care of this).